static/app/components/onboarding/productSelection.tsx

```diff
@@ -384,6 +392,24 @@ export function ProductSelection({
     [defaultProducts, organization, router, urlProducts]
   );
 
+  const handleToggleLoader = useCallback(() => {
+    if (!showLoader === false && platform) {
+      trackAnalytics('onboarding.js_loader_npm_docs_shown', {
+        organization,
+        platform,
+        project_id: projectId,
+      });
+    }
+
+    router.replace({
+      pathname: router.location.pathname,
+      query: {
+        ...router.location.query,
+        showLoader: !showLoader,
+      },
+    });
+  }, [router, showLoader, platform, projectId, organization]);
```

Comment:
You could put this + accessing the query param into a hook which can be used here and in `platform.tsx`:
```ts
const [
  showLoaderOnboarding,
  toggleLoaderOnboarding
] = useLoaderOnboardingState()
```
Then it is ensured that both components access the same query parameter.

Reply:
that is a very good idea! will implement that 🙏